src/GraphQL/Resolver/DomainContentResolver.php

```diff
@@ -198,6 +353,67 @@ public function resolveContentName(ContentInfo $contentInfo)
         return $this->repository->getContentService()->loadContentByContentInfo($contentInfo)->getName();
     }
 
+    private function getInputFieldValue($input, Repository\Values\ContentType\FieldDefinition $fieldDefinition)
+    {
+        $supportedInputTypes = ['ezstring', 'ezimage', 'eztext', 'ezrichtext', 'ezauthor'];
+        if (!in_array($fieldDefinition->fieldTypeIdentifier, $supportedInputTypes)) {
+            throw new UnsupportedFieldTypeException($fieldDefinition->fieldTypeIdentifier, 'input');
+        }
+
+        if ($fieldDefinition->fieldTypeIdentifier === 'ezauthor') {
+            $authors = [];
+            foreach ($input[$fieldDefinition->identifier] as $authorInput) {
+                $authors[] = new FieldType\Author\Author($authorInput);
+            }
+            $fieldValue = new FieldType\Author\Value($authors);
+        } elseif ($fieldDefinition->fieldTypeIdentifier === 'ezimage') {
+            $fieldInput = $input[$fieldDefinition->identifier];
+
+            if (!$fieldInput['file'] instanceof UploadedFile) {
+                return null;
+            }
+            $file = $fieldInput['file'];
+            $fieldValue = new FieldType\Image\Value([
+                'alternativeText' => $fieldInput['alternativeText'] ?? '',
+                'fileName' => $file->getClientOriginalName(),
+                'inputUri' => $file->getPathname(),
+                'fileSize' => $file->getSize(),
+            ]);
+        } elseif ($fieldDefinition->fieldTypeIdentifier === 'ezrichtext') {
+            $format = $input[$fieldDefinition->identifier]['format'];
+            $input = $input[$fieldDefinition->identifier]['input'];
+
+            if ($format === 'html') {
```

Comment:
Refactor this with `RichTextFormatConverter` objects (that handle both input and output ([EZP-30212](https://jira.ez.no/browse/EZP-30212))).